cnmc_client/client.py

```diff
@@ -107,6 +107,9 @@ def fetch_massive(self, cups, file_type, as_csv=False, wait=0):
         number_of_cups = len(cups)
         chunk_indexes = [x*CUPS_CHUNK_SIZE for x in range(0, number_of_cups/CUPS_CHUNK_SIZE + 1)]
 
+        if chunk_indexes[len(chunk_indexes) - 1:] == [len(cups)]:
+            chunk_indexes = chunk_indexes[:-1]
+
```

Comment:
Maybe I don't understand the initial code.
What is `chunk_indexes` and `chunk_block`?

To split lists in chunks I think this could be as following:

1. Calculating chunks on demand (more eficient in memory)
```python
results = []
start = 0
cups_block = cups[start:start + CUPS_CHUNK_SIZE]
while cups_block:
    if start > 0:
        time.sleep(wait)
    result = self.fetch(cups_block, file_type, as_csv)
    results.append(result)
    start += CUPS_CHUNK_SIZE
    cups_block = cups[start:start + CUPS_CHUNK_SIZE]
return results
```
2. Pre calculated chunks and returning a lists of lists
```python
results = []
cups_blocks = [cups[x:x + CUPS_CHUNK_SIZE] for x in range(0, len(cups), CUPS_CHUNK_SIZE)]
for idx, cups_block in enumerate(cups_blocks):
    if idx > 1:
        time.sleep(wait)
    result = self.fetch(cups_block, file_type, as_csv)
    results.append(result)
```

This will avoid to *pre calculate* the chunk indexes.
We can improve a little bit more and doing the `time.sleep` only if `start > 0` or `idx > 1` because we don't need to sleep the first time.